src/responder/common/cache_req/cache_req.c

```diff
@@ -1019,10 +1043,12 @@ static void cache_req_search_domains_done(struct tevent_req *subreq)
     case EAGAIN:
         break;
     default:
-        if (ret == ENOENT && state->cr->data->propogate_offline_status
-                && !state->dp_success) {
+        if (cache_req_dp_contacted(state)
+            && ret == ENOENT
+            && !state->dp_success
+            && state->cr->data->propogate_offline_status) {
```

Comment:
Why is this repeated twice (here and at line 1009)?

Reply:
If I understand the code completely, the code on 1008 is for lookup when domain was specified in the input. But it looks like it can be removed since we should get ENOENT from `cache_req_search_domains_next`.

Reply:
And removed the code on 1009, it looks like it works correctly without it.

Reply:
This actually created issues in unit test, I will look into it and see if it is something easy to fix.

Reply:
Fixing the tests would take a significant effort, out of scope for this PR. Let's keep the check.

Reply:
ok